docker_cli/_image.py

```diff
@@ -0,0 +1,451 @@
+        Use 'docker inspect' to retrieve a piece of information about the
+        image.
+
+        Parameters
+        ----------
+        format : str, optional
+            The value to be requested by the --format argument, or None.
+            Defaults to None.
+
+        Returns
+        -------
+        str
+            The string returned by the docker inspect command.
+        """
+        cmd = ["docker", "inspect"]
+        if format:
+            cmd += [f"-f={format}"]
+        cmd += [self._id]
+
+        inspect_result = run(
+            cmd,
+            capture_output=True,
+            text=True,
+            check=True
+        )
+
+        output_text = inspect_result.stdout
+        return output_text
+
+    def run(
+        self,
+        command: str,
+        *,
+        stdout: Optional[Union[io.TextIOBase, int]] = None,
+        stderr: Optional[Union[io.TextIOBase, int]] = None,
+        interactive: bool = False,
+        network: str = "host"
+    ) -> str:
+        """
+        Run the given command on a container.
+
+        .. warning::
+            This method does not work correctly if the image built does not have
+            bash installed.
+
+        Parameters
+        ----------
+        cmd : str
+            The desired command, in Unix shell syntax.
+        stdout : io.TextIOBase or special value, optional
+            For a description of valid values, see :func:`subprocess.run`.
+        stderr : io.TextIOBase or special value, optional
+            For a description of valid values, see :func:`subprocess.run`.
+        interactive : bool, optional
+            A boolean describing whether or not to run this command in
+            interactive mode or not. Defaults to False.
+        network : str, optional
+            The name of the network. Defaults to "host".
+
+        Returns
+        -------
+        str, optional
+            The output of the process, if `stdout` == PIPE.
+
+        Raises
+        -------
+        CommandNotFoundError:
+            When a command is attempted that is not recognized on the image.
+        """
+        cmd = ["docker", "run", f"--network={network}", "--rm"]
+        if interactive:
+            cmd += ["-i"]
+            if stdin.isatty():
+                cmd += ["--tty"]
+        cmd += [self._id, "bash"]
+        cmd += ["-ci"] if interactive else ["-c"]
+        cmd += split(f"'{command}'")
+
+        try:
+            result = run(
+                cmd,
+                text=True,
+                stdout=stdout,  # type: ignore
+                stderr=stderr,  # type: ignore
+                check=True
+            )
+            retval = result.stdout
+        except CalledProcessError as err:
+            if err.returncode == 127:
+                raise CommandNotFoundError(split(command)[0]) from err
+            else:
+                raise
+        return retval
+
+    def drop_in(
+        self,
+        network: str = "host"
+    ) -> None:
```

Comment:
This function signature isn't too long so let's unwrap it.

```suggestion
    def drop_in(self, network: str = "host") -> None:
```